test/dictionary_test.pl

```diff
@@ -4,6 +4,35 @@
 % To run the tests:
 % ?- run_tests.
 
+test(hasKey,[nondet]) :- % used the [nondet] option to supress the error "Test succeeded with choicepoint"
```

Comment:
All the predicates need to be deterministic (otherwise the vault might perform two different actions for the same command). A better way to fix the error is to make sure the same input doesn't fit two cases, or use fails `!` to stop Prolog from exploring more cases
```suggestion
test(hasKey) :- 
```